torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -147,23 +145,41 @@ def __init__(
         parameters_as_bucket_view: bool = False,
         **default: Any,
     ):
+        # Validate the type and length of `params`
+        # NOTE: While `Optimizer` allows `params` to be an iterable of `dict`s,
+        # ZeRO strictly requires `params` to be an iterable of `torch.Tensor`s
+        if (isinstance(params, torch.Tensor)):
+            raise TypeError("params argument should be an iterable of "
+                            "Tensors, but got " + torch.typename(params))
+        try:
+            self._all_params = list(params)
```

Comment:
Does this mean it now allows generators? If yes, shall we add a test?

Reply:
Yes, I can add an explicit test. I want to note though that passing `params` as a generator is already _implicitly_ tested in the existing test suite whenever a test passes `model.parameters()` to the constructor since that should be a generator.

Reply:
Fixed.